packages/core/src/combinator/timeslice.js

```diff
@@ -4,110 +4,135 @@
 
 import Pipe from '../sink/Pipe'
 import { disposeAll } from '@most/disposable'
+import { map } from '@most/prelude'
+import { empty, isCanonicalEmpty } from '../source/empty'
 import { join } from './chain'
+import { take } from './slice'
+import { mergeArray } from './merge'
+
+// Time bounds
+// TODO: Move to own module
+const timeBounds = (min, max) =>
+  ({ min, max })
+
+const minBounds = min =>
+  timeBounds([min], [])
+
+const maxBounds = max =>
+  timeBounds([], [max])
+
+const mergeTimeBounds = (b1, b2) =>
+  timeBounds(b1.min.concat(b2.min), b1.max.concat(b2.max))
+
+// Interpret time bounds
+const getStartingMin = ({ min }) =>
```

Comment:
No reason to have a `get` prefix. `startingMin` is fine.

Reply:
:+1: I updated naming in e4edf53 based on your suggestions.